tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java

```diff
@@ -75,70 +72,75 @@ public void testRfc2617() throws Exception {
+        MessageDigest md = MessageDigest.getInstance("MD5");
+
+        A1 = "Mufasa" + ":" + "testrealm@host.com" + ":" + "Circle Of Life";
+        HashA1 = encode(A1, md);
+        String A2;
+        A2 = "GET" + ":" + uri;
+        String HashA2 = encode(A2, md);
+        String combo, finalHash;
+        combo = HashA1 + ":" + nonce + ":" + nc + ":" +
+                cnonce + ":auth:" + HashA2;
+        finalHash = encode(combo, md);
+
+        return finalHash;
+    }
+
+    private String encode(String src, MessageDigest md) {
+        char[] charArray = {
+                '0', '1', '2', '3', '4', '5', '6', '7',
+                '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
+        };
+        md.update(src.getBytes(StandardCharsets.ISO_8859_1));
+        byte[] digest = md.digest();
+        StringBuilder res = new StringBuilder(digest.length * 2);
+        for (byte b : digest) {
+            int hashChar = ((b >>> 4) & 0xf);
+            res.append(charArray[hashChar]);
+            hashChar = (b & 0xf);
+            res.append(charArray[hashChar]);
+        }
+        return res.toString();
+    }
+
     @Test
-    public void testRfc2617EncodedPath() throws Exception {
+    public void testRfc2617b() throws Exception {
         mockDigestNonce("AAAAAQABsxiWa25/kpFxsPCrpDCFsjkTzs/Xr7RPsi/VVN6faYp21Hia3h4=");
         Map<String, Object> props = new HashMap<>();
         props.put(CONFIG_REALM, "testrealm@host.com");
-        props.put("org.wildfly.security.http.validate-digest-uri", "true");
+        props.put("org.wildfly.security.http.validate-digest-uri", "false");
         HttpServerAuthenticationMechanism mechanism = digestFactory.createAuthenticationMechanism(DIGEST_NAME, props, getCallbackHandler("Mufasa", "testrealm@host.com", "Circle Of Life"));
-
-        String path = "/dir/foo%2Fr/index.html?foo=b%2Fr";
-        String uri = "http://localhost" + path;
         TestingHttpServerRequest request2 = new TestingHttpServerRequest(new String[] {
                 "Digest username=\"Mufasa\",\n" +
                         "                 realm=\"testrealm@host.com\",\n" +
                         "                 nonce=\"dcd98b7102dd2f0e8b11d0f600bfb0c093\",\n" +
-                        "                 uri=\"" + path + "\",\n" +
+                        "                 uri=\"/dir/index.html\",\n" +
                         "                 qop=auth,\n" +
                         "                 nc=00000001,\n" +
                         "                 cnonce=\"0a4f113b\",\n" +
-                        "                 response=\"cc3261565007973c9d647333822091ad\",\n" +
+                        "                 response=\"" + computeDigest("/dir/index.html", "dcd98b7102dd2f0e8b11d0f600bfb0c093", "0a4f113b", "00000001") + "\",\n" +
                         "                 opaque=\"00000000000000000000000000000000\",\n" +
                         "                 algorithm=MD5"
-        }, new URI(uri));
```

Comment:
@miador testRfc2617EncodedPath test method was removed, maybe you need to rebase on 1.x?